### PR TITLE
(IRO-1793) Introduce search autocomplete component.

### PR DESCRIPTION
**Pull request by azhurauski-sc**

### CODE REVIEW

File: components/SearchAutocomplete.tsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+                onClose()
+              }
+            }}
+          >
+            {renderOption(option)}
+          </Box>
+        ))}
+      </>
+    )
+  }
```
Comment (brekk) [inline components/SearchAutocomplete.tsx:109]: This feels a lot like a standalone component.

Can you pass the stateful props in and set this up as a `<Group>` component?

File: stories/Search.stories.tsx [hunk 2]
```diff
@@ -29,10 +34,93 @@ export const SearchInput: ComponentStory<FC> = () => (
...
+
+  while (result.length < parts - 1) {
+    const startPosition = result.length * blockLength
+    result.push(hash.slice(startPosition, startPosition + chars))
+  }
+
+  result.push(hash.slice(hash.length - chars, hash.length))
+
+  return result.join('...')
+}
```
Comment (brekk) [inline stories/Search.stories.tsx:79]: You may want to look at this CSS only approach we're using in `website-testnet` -- https://github.com/iron-fish/website-testnet/blob/master/components/user/EventRow.module.css#L51-L56

The benefits there are that you still have the raw data (which in that case was needed for copy-ability) but visually it looks cleaner.

File: stories/Search.stories.tsx [hunk 3]
```diff
@@ -29,10 +34,93 @@ export const SearchInput: ComponentStory<FC> = () => (
...
+export const SearchAutocompleteInput: ComponentStory<FC> = () => (
+  <Stack>
+    <Box display="column" width="18.75rem">
+      <h4>Autocomplete search</h4>
+      <SearchAutocomplete
+        InputProps={{ placeholder: 'Search' }}
+        variant="nav_search"
+        inputLeftElement={<IconSearch />}
+        options={[...new Array(10)].map((item, index) => {
+          const value = `000045645665456434634564456400004564566545643463456445644564456${index}`
```
Comment (brekk) [inline stories/Search.stories.tsx:102]: This is just example data?

File: components/SearchAutocomplete.tsx [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+  )
+
+  const renderGroup = (groupName: string, groupOptions: SearchOptionType[]) => {
+    return (
+      <>
+        {groupName ? (
+          <Box w="100%" sx={styles?.groupTitleWrapper}>
+            {renderGroupTitle(groupName)}
+          </Box>
+        ) : null}
```
Comment (ndemidovich-sc) [inline components/SearchAutocomplete.tsx:90]:
```suggestion
        {groupName && (
          <Box w="100%" sx={styles?.groupTitleWrapper}>
            {renderGroupTitle(groupName)}
          </Box>
        )}
```

File: components/SearchAutocomplete.tsx (hunk 1 above)
Comment (ndemidovich-sc) [reply components/SearchAutocomplete.tsx:109]: Also I suggest to use something like next:
```tsx
<Group label="A">
  <Option label="A1"/>
  <Option label="A2"/>
</Group>
<Group label="B">
  <Option label="B1"/>
  <Option label="B2"/>
</Group>
```

File: components/SearchAutocomplete.tsx [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+  value = null,
+  InputProps = {},
+  options = [],
+  emptyOption = 'No matches',
+  renderOption = option => <Box>{option.label}</Box>,
+  onSelectOption = () => void 0,
+  groupOptionsBy = option => option.object,
+  renderGroupTitle = title => <Text>{title}</Text>,
+  inputLeftElement,
+  ...props
```
Comment (ndemidovich-sc) [inline components/SearchAutocomplete.tsx:58]: Can we refactor it to more clear props structure?

**Review by ndemidovich-sc — CHANGES_REQUESTED** [review]

File: stories/Search.stories.tsx (hunk 3 above)
Comment (azhurauski-sc) [reply stories/Search.stories.tsx:102]: yes

**Review by ndemidovich-sc — APPROVED** [review]

**Review by brekk — APPROVED** [review]

**Pull request merged**